osu.Game/Beatmaps/Drawables/DifficultyIcon.cs

```diff
@@ -58,11 +61,14 @@ public virtual Bindable<StarDifficulty> Current
         /// Creates a new <see cref="DifficultyIcon"/>. Will use provided beatmap's <see cref="BeatmapInfo.StarRating"/> for initial value.
         /// </summary>
         /// <param name="beatmap">The beatmap to be displayed in the tooltip, and to be used for the initial star rating value.</param>
+        /// <param name="mods">The mods type beat</param>
         /// <param name="ruleset">An optional ruleset to be used for the icon display, in place of the beatmap's ruleset.</param>
-        public DifficultyIcon(IBeatmapInfo beatmap, IRulesetInfo? ruleset = null)
+        public DifficultyIcon(IBeatmapInfo beatmap, IRulesetInfo? ruleset = null, Mod[]? mods = null)
```

Comment:
Changing this class means changing every usage of it and not just multiplayer. Notably this icon & its tooltip are used on song select.

![osu_2024-01-17_13-44-26](https://github.com/ppy/osu/assets/20418176/9616c4ae-c2bf-47b9-8038-b94c5d461b2b)

I'm not super sure we want all of that stuff there. Certainly wasn't discussed.

Reply:
Also missed this, my bad. I think I'll just add another bool argument which defaults to false. It would be used to display all of this information, or just the star rating.